apps/passport-client/src/dispatch.ts

```diff
@@ -656,9 +665,12 @@ let syncInProgress = false;
 let skippedSyncUpdates = 0;
 
 /**
- * Does the real work of sync(), inside of reentrancy protection.
- * Returns the changes to be made to AppState.  If changes were made,
- * this function should be run again.
+ * Does the real work of sync(), inside of reentrancy protection.  If action
+ * is needed, this function takes one action and returns, expecting to be
+ * run again until no further action is needed.
+ *
+ * Returns the changes to be made to AppState.  If the result is defined, this
+ * function should be run again.
```

Comment:
worth commenting what the caller is meant to do with the value that is returned from `doSync`